Codesytle update

PreisTo opened the pull request:
- updated flake8, isort and black configuration for more strict and general codestyle, following [this](https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html)
- updated versioneer files sing versioneer v0.29 (fixed multiple flake8 warnings)
- fixed pkg_resources deprecation
- black all files
- fixed flake8 errors
- docstring fromatting using [docformatter](https://github.com/PyCQA/docformatter)

ndilalla commented on astromodels/core/model_parser.py:
```diff
@@ -596,10 +597,10 @@ def _parse_sky_direction(self, sky_direction_definition):
             par_parser = ParameterParser("l", sky_direction_definition["l"])
 
-            l = par_parser.get_variable()
+            lon = par_parser.get_variable()
```
Why is it changing variable names? I don't see the need to, honestly, as l and b are galactic coordinates

PreisTo replied:
flake8 rule [E741](https://www.flake8rules.com/rules/E741.html) states to not use "l","I" or "O" as variable names  - I've only changed "internal" appearances and did not mess with  function/class inputs. But I think excluding this rule is  the best solution. This is also the reason for the CI failing :upside_down_face:

ndilalla commented:
Thanks @PreisTo for opening this PR. I see that the tests are failing at the linting step. Can you take a look?

PreisTo commented:
> Thanks @PreisTo for opening this PR. I see that the tests are failing at the linting step. Can you take a look?
Thanks @ndilalla for the quick response :) 
I've forgotten that having remaining flake8 warnings will make the CI fail, in this case the same error as before is this reason:
```bash
./astromodels/core/polarization.py:128:24: E741 ambiguous variable name 'I'
./astromodels/core/sky_direction.py:27:43: E741 ambiguous variable name 'l'
./astromodels/functions/functions_1D/apec.py:329:39: E741 ambiguous variable name 'O'
```

I've undone all my changes renaming "l" and "b" and instead added E741 to the ignore list

PreisTo commented:
I've fixed some bugs/typos causing the CI to fail - now all jobs from CI are passing at least on my repo :) 

Also I've removed some future deprecations (astropy `in_units` and Python 3.13 concerning locals() when generating the xspec models (see [PEP 558](https://peps.python.org/pep-0558/))) which are both downward compatible until at least 3.9

ndilalla commented:
Thank you! @PreisTo I will check this PR and merge it soon
